src/internal.c

```diff
@@ -20782,6 +20783,18 @@ int ProcessReplyEx(WOLFSSL* ssl, int allowSocketErr)
                                 ssl->buffers.inputBuffer.buffer,
                                 &ssl->buffers.inputBuffer.idx,
                                 ssl->buffers.inputBuffer.length);
+                            if (ret == 0 && ssl->options.dtlsStateful) {
+                                ret = Dtls13UpdateWindow(ssl);
+                                if (ret != 0) {
+                                    WOLFSSL_ERROR(ret);
+                                    return ret;
+                                }
+                                ret = Dtls13RecordRecvd(ssl);
+                                if (ret != 0) {
```

Comment:
What if we change `Dtls13RecordRecvd` so that it doesn't add a new `Dtls13RecordNumber` if we already have the epoch/seq number already stored in `ssl->dtls13Rtx.seenRecords`?

Reply:
It looks rather counter-intuitive to me as replay detection is done per record. You should also update the window when you have a record with just a fragment: in this case Dtls*DoHandshakeMsg is not invoked at all. What's the main drive to move it after the message? Is it because the window is updated in stateless code? can we reset it if we don't go stateful?

Reply:
I think this is still a critical issue in the PR. Besides complicating/duplicating the code, with the actual approach the window will be updated just for the last record for a multi-records-single-message interaction, and multiple times for a single-record-multiple-messages type of packet.  Consider that RecordRecv must be per record for retransmission to work correctly. 

Can you consider changing as little as possible of the record window code path and resetting the window while we are in stateless mode?